contracts/PlasmaMVP.sol

```diff
@@ -409,12 +412,18 @@ contract PlasmaMVP {
         //
         // For a fee mismatch, the state becomes `NonExistent` so that the exit can be reopened.
         // Otherwise, `Challenged` so that the exit can never be opened.
-        if (firstInput && exit_.committedFee != txList[14].toUintStrict()) {
+        if (firstInput && exit_.committedFee != txList[14].toUintStrict() && confirmSignature.length == 0) {
             bytes memory sig = sigList[0].toBytes();
             recoveredAddress = txHash.recover(sig);
             require(sig.length == 65 && recoveredAddress != address(0) && exit_.owner == recoveredAddress);
 
-            exit_.state = ExitState.NonExistent;
+            delete exit_.amount;
```

Comment:
because `exit_` is a storage pointer, `delete exit_` might suffice. Can you check this?

Reply:
`: Unary operator delete cannot be applied to type struct PlasmaMVP.exit storage pointer`